frame/transaction-storage/src/lib.rs

```diff
@@ -0,0 +1,354 @@
+		pub(super) fn check_proof(
+			origin: OriginFor<T>,
+			proof: Option<TransactionStorageProof>,
+		) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+			let number = <frame_system::Pallet<T>>::block_number();
+			let period = <StoragePeriod<T>>::get().unwrap_or(DEFAULT_STORAGE_PERIOD.into());
+			let target_number = number.saturating_sub(period);
+			if target_number.is_zero() {
+				ensure!(proof.is_none(), Error::<T>::UnexpectedProof);
+				return Ok(().into())
+			}
+			let transaction_count = <TransactionCount<T>>::get(target_number);
+			let mut total_chunks = 0;
+			for t in 0 .. transaction_count {
+				match <TransactionRoots<T>>::get(target_number, t) {
+					Some(info) => total_chunks +=
+						((info.size as u64 + CHUNK_SIZE as u64 - 1) / CHUNK_SIZE as u64) as u32,
+					None => break,
+				}
+			}
+			if total_chunks == 0 {
+				ensure!(proof.is_none(), Error::<T>::UnexpectedProof);
+				return Ok(().into())
+			}
+			let proof = proof.ok_or_else(|| Error::<T>::MissingProof)?;
+			let parent_hash = <frame_system::Pallet<T>>::parent_hash();
+			let selected_chunk_index = random_chunk(parent_hash.as_ref(), total_chunks);
+			total_chunks = 0;
+			let mut t = 0;
+			let (info, chunk_index) = loop {
+				match <TransactionRoots<T>>::get(target_number, t) {
+					Some(info) => {
+						let chunks = ((info.size as u64 + CHUNK_SIZE as u64 - 1) / CHUNK_SIZE as u64) as u32;
+						if total_chunks + chunks >= selected_chunk_index {
+							break (info, selected_chunk_index - total_chunks)
+						}
+						total_chunks += chunks;
+					},
+					None => Err(Error::<T>::MissingStateData)?,
+				}
+				t += 1;
+			};
+
+			ensure!(
+				sp_io::trie::blake2_256_verify_proof(
+					info.chunk_root,
+					&proof.proof,
+					&encode_index(chunk_index),
+					&proof.chunk,
+				),
+				Error::<T>::InvalidProof
+			);
+			Self::deposit_event(Event::ProofChecked);
+			Ok(().into())
+		}
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Stored data under specified index.
+		Stored(u32),
+		/// Renewed data under specified index.
+		Renewed(u32),
+		/// Storage proof was successfully checked.
+		ProofChecked,
+	}
+
+	/// Collection of extrinsic storage roots for the current block.
+	#[pallet::storage]
+	#[pallet::getter(fn transaction_roots)]
+	pub(super) type TransactionRoots<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::BlockNumber,
+		Blake2_128Concat,
+		u32,
+		TransactionInfo,
+		OptionQuery,
+	>;
+
+	/// Count indexed chunks for each block.
+	#[pallet::storage]
+	pub(super) type TransactionCount<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::BlockNumber,
+		u32,
+		ValueQuery,
+	>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn byte_fee)]
+	/// Storage fee per byte.
+	pub(super) type ByteFee<T: Config> = StorageValue<_, BalanceOf<T>>;
+
+	#[pallet::storage]
+	pub(super) type Counter<T: Config> = StorageValue<_, u32>;
```

Comment:
```suggestion
	pub(super) type Counter<T: Config> = StorageValue<_, u32, ValueQuery>;
```

I think with this you'll get the default for `u32` in case of nothing being there.